src/test/builders.test.ts

```diff
@@ -0,0 +1,105 @@
+import { createBuilderClass, createLink } from './builders/base';
+import assert = require('assert');
+
+interface IGrandChild {
+	attr: number;
+}
+
+const GrandChildBuilder = createBuilderClass<IGrandChild>()({
+	attr: {default: 10},
+});
+
+interface IChild {
+	name: string;
+	grandchild: IGrandChild;
+}
+
+const ChildBuilder = createBuilderClass<IChild>()({
+	name: {default: ''},
+	grandchild: {linked: GrandChildBuilder},
+});
+
+const b = new ChildBuilder();
+b.grandchild((gc) => gc.attr(20));
+
+interface IParent {
+	aStringProp: string;
+	aNumberProp: number;
+	aBooleanProp: boolean;
+	aChildProp: IChild;
+}
+
+const ParentBuilder = createBuilderClass<IParent>()({
+	aStringProp: {default: 'abc'},
+	aNumberProp: {default: 123},
+	aBooleanProp: {default: true},
+	aChildProp: {linked: ChildBuilder},
+});
+
+describe('Builders', function () {
```

Comment:
Fun fact: almost all of the actual complexity of the Builder tests is in getting them to typecheck. :stars: